v1/ast/parser.go

```diff
@@ -0,0 +1,2742 @@
+// Copyright 2020 The OPA Authors.  All rights reserved.
+// Use of this source code is governed by an Apache2
+// license that can be found in the LICENSE file.
+
+package ast
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"math/big"
+	"net/url"
+	"regexp"
+	"sort"
+	"strconv"
+	"strings"
+	"unicode/utf8"
+
+	"gopkg.in/yaml.v3"
+
+	"github.com/open-policy-agent/opa/v1/ast/internal/scanner"
+	"github.com/open-policy-agent/opa/v1/ast/internal/tokens"
+	astJSON "github.com/open-policy-agent/opa/v1/ast/json"
+	"github.com/open-policy-agent/opa/v1/ast/location"
+)
+
+var RegoV1CompatibleRef = Ref{VarTerm("rego"), StringTerm("v1")}
+
+// RegoVersion defines the Rego syntax requirements for a module.
+type RegoVersion int
+
+const DefaultRegoVersion = RegoV1
+
+const (
+	RegoUndefined RegoVersion = iota
```

Comment:
This rego-version is the default value for any `ast.RegoVersion` struct field, e.g. `ast.ParserOptions.RegoVersion` and `ast.Module.regoVersion`. Which allows us to select whatever rego-version we should actually default to based on the user's imports.
E.g.: 
* https://github.com/open-policy-agent/opa/blob/31b5f173c4f88b6e1d5f2d7247b8e88fe934888a/sdk/opa.go#L33
* https://github.com/open-policy-agent/opa/blob/8b531cafba9e4e54c47aa51678d055a83fb7a41d/v1/ast/compile.go#L1757